plugins/database/mongodb/connection_producer.go

```diff
@@ -206,3 +262,29 @@ func (c *mongoDBConnectionProducer) getTLSAuth() (opts *options.ClientOptions, e
 	opts.SetTLSConfig(tlsConfig)
 	return opts, nil
 }
+
+func (c *mongoDBConnectionProducer) getTimeoutOpts() (opts *options.ClientOptions, err error) {
```

Comment:
nit: Getters [don't have to include](https://golang.org/doc/effective_go#Getters) the `get` prefix on func names. I know we're not super consistent on this and it's less severe for private methods, but might be good to avoid this going forward.

Reply:
Agreed that we should try to be consistent about this, however I argue that this isn't a getter. This is creating a new `ClientOptions` instance.

Reply:
I think that `c.timeoutOpts()` or `c.clientTimeoutOpts()` would still make sense even so.

Reply:
Updated